joern-cli/frontends/rubysrc2cpg/src/main/java/io/joern/rubysrc2cpg/parser/RubyLexerBase.java

```diff
@@ -0,0 +1,98 @@
+package io.joern.rubysrc2cpg.parser;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.Token;
+import static io.joern.rubysrc2cpg.parser.RubyLexer.*;
+
+/** Aggregates auxiliary features to RubyLexer in a single place. */
+public abstract class RubyLexerBase extends Lexer {
+    protected RubyLexerBase(CharStream input) {
+        super(input);
+    }
+
+    /* When encountering '/', we need to decide whether this is a binary operator (e.g. `x / y`) or
+     * a regular expression delimiter (e.g. `/(eu|us)/`) occurrence. Our approach is to look at the
+     * previously emitted token and decide accordingly.
+     */
+
+    /** The previously emitted token (in DEFAULT_CHANNEL.) */
+    private Token previousToken = null;
+
+    // Same original behaviour, just updating `previousToken`.
+    @Override
+    public Token nextToken() {
+        Token token = super.nextToken();
+        if (token.getChannel() == Token.DEFAULT_CHANNEL){
+            previousToken = token;
+        }
+        return token;
+    }
+
+    /** To be invoked when encountering `/`, deciding if it should emit a
+     * `REGULAR_EXPRESSION_START` token. */
+    protected boolean couldBeRegexStart(){
```

Comment:
just a nit, but since this method is used to determine whether we are at a regex start or not I would expect the method to be named something like `isStartOfRegex`.